nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1324 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import shutil
+import warnings
+
+import numpy as np
+import pyroomacoustics as pra
+import soundfile as sf
+import torch
+from gpuRIR import att2t_SabineEstimator, beta_SabineEstimation, simulateRIR, t2n
+from pyroomacoustics.directivities import CardioidFamily, DirectionVector, DirectivityPattern
+from scipy.signal import convolve
+from scipy.signal.windows import cosine, hamming, hann
+from scipy.stats import halfnorm
+
+from nemo.collections.asr.parts.utils.manifest_utils import (
+    create_manifest,
+    create_segment_manifest,
+    read_manifest,
+    write_ctm,
+    write_manifest,
+    write_text,
+)
+from nemo.collections.asr.parts.utils.speaker_utils import labels_to_rttmfile
+from nemo.utils import logging
+
+
+class MultiSpeakerSimulator(object):
+    """
+    Multispeaker Audio Session Simulator.
+
+    Args:
+        cfg: OmegaConf configuration loaded from yaml file.
+    """
+
+    def __init__(self, cfg):
+        self._params = cfg
+        # internal params
+        try:
+            self._manifest = read_manifest(self._params.data_simulator.manifest_path)
+        except:
+            raise Exception("Manifest file could not be opened")
+        self._sentence = None
+        self._text = ""
+        self._words = []
+        self._alignments = []
+        # keep track of furthest sample per speaker to avoid overlapping same speaker
+        self._furthest_sample = [0 for n in range(0, self._params.data_simulator.session_config.num_speakers)]
+        # use to ensure overlap percentage is correct
+        self._missing_overlap = 0
+        # creating manifests during online data simulation
+        self.base_manifest_filepath = None
+        self.segment_manifest_filepath = None
+        # variable speaker volume
+        self._volume = None
+        self._check_args()  # error check arguments
+
+    def _check_args(self):
```

Comment:
since you need gpuRIR or pyroomacoustics only if user selects RIR, add imports here checking for those argument values. That way user can skip installing these libraries if they don;t want to

Reply:
I added this at the start of the code:
```
try:
    import pyroomacoustics as pra
    from pyroomacoustics.directivities import CardioidFamily, DirectionVector, DirectivityPattern
    PRA = True
except ImportError:
    PRA = False
try:
    from gpuRIR import att2t_SabineEstimator, beta_SabineEstimation, simulateRIR, t2n
    GPURIR = True
except ImportError:
    GPURIR = False
```
And then only checked whether the desired toolkit was installed in _check_args.